ccc-js/src/main/javascript/package-res/lib/protovis.js

```diff
@@ -4521,14 +4520,17 @@ DateComponent.prototype.floor = function(d, options) {
 };
 
 DateComponent.prototype.floorMultiple = function(d, n, options) {
-  var first = this.first(d, options),
-      delta = this.get(d) - first; // base in units
-
+  var alignVal = pv.get(options, 'alignmentValue');
+  var alignRef = alignVal ? this.get(alignVal) : this.first(d, options);
+  var delta = this.get(d) - alignRef; // base in units
+  
   if(delta) {
     var M = n * this.mult,
         offset = Math.floor(delta / M) * M; // offset in base units
-
-    this.set(d, first + offset);
+    
+    alignRef += offset >= 0 ? getDays(d) - M : offset;
```

Comment:
This function is used for all date components: year, month, week, day, ...
`getDays` isn't general enough.
Also, probably what matters is whether if `alignRef + offset` is negative or not, not directly `offset`, no?

Reply:
Please take a look at:
https://github.com/webdetails/ccc/pull/332/files#diff-256db534a86ae1bcf174f30fe5e411f7R4532

It was the best way i found to deal with tick alignments to different years/ months:
. when we have a tick align value, we'll floor it to the the first multiple's lower than  the given date. we´ll then set the date respecting the alignment date year/ month.
. i offloaded some of it the date component set method. that way, each component can (when needed) to deal with on their own way.